signer/src/stacks/wallet.rs

```diff
@@ -495,4 +559,82 @@ mod tests {
 
         assert_eq!(wallet1.address(), wallet2.address())
     }
+
+    /// Here we test that we can load a SignerWallet from storage. To do
+    /// that we:
+    /// 1. Generate and store random bitcoin and stacks blockchains.
+    /// 2. Create a random wallet.
+    /// 3. Generate a rotate-keys transaction object using the details of
+    ///    the random wallet from (2).
+    /// 4. Attempt to load the wallet from storage. This should return
+    ///    essentially the same wallet from (2). The only difference is
+    ///    that the nonce in the loaded wallet is fetched from the
+    ///    "stacks-node" (in this test it just returns a nonce of zero).
+    #[tokio::test]
+    async fn loading_signer_wallet_from_context() {
+        let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+
+        let ctx = TestContext::builder()
+            .with_in_memory_storage()
+            .with_mocked_clients()
+            .build();
+        let db = ctx.get_storage_mut();
+
+        // Create a blockchain. We do not generate any withdrawal or
+        // deposit requests, so we do not need to specify the signer set.
+        let test_params = crate::testing::storage::model::Params {
+            num_bitcoin_blocks: 10,
+            num_stacks_blocks_per_bitcoin_block: 0,
```

Comment:
Ahh, I believe `(1..0)` is an empty range.

Reply:
it seems to be still considering the initial value though:
```
# dbg!((1..0).fold(vec![1], |a, _| a));
[src/main.rs:5:5] (1..0).fold(vec![1], |a, _| a) = [
    1,
]
```

Reply:
Oh, right, well that settles it. My mistake! 

Also, I did not take in the confusing part about this test, which is that I use Stacks blocks but ask for zero stacks blocks to be generated. I'll fix this in another PR.

> To avoid surprises, should we validate `num_stacks_blocks_per_bitcoin_block > 0` or avoid creating a stacks block if the param is zero?

I would think that we should avoid creating Stacks blocks if we ask for zero. For this code, I should probably just update the number of stacks blocks that are needed to avoid confusion. I'll make sure that it is added in.